samples/kotlin-android-app-destinations/src/main/java/com/segment/analytics/destinations/plugins/FirebaseDestination.kt

```diff
@@ -0,0 +1,236 @@
+                "name" to Param.ITEM_NAME,
+                "price" to Param.PRICE,
+                "quantity" to Param.QUANTITY
+        )
+    }
+
+    override fun setup(analytics: Analytics) {
+        super.setup(analytics)
+
+        firebaseAnalytics = FirebaseAnalytics.getInstance(context)
+    }
+
+    override fun identify(payload: IdentifyEvent): BaseEvent? {
+        var returnPayload = super.identify(payload)
+
+        firebaseAnalytics.setUserId(payload.userId)
+
+        payload.traits.let {
+            for ((traitKey, traitValue) in it) {
+                val updatedTrait = makeKey(traitValue.toString())
+                firebaseAnalytics.setUserProperty(traitKey, updatedTrait)
+            }
+        }
+
+        return returnPayload
+    }
+
+    override fun track(payload: TrackEvent): BaseEvent? {
+        var returnPayload = super.track(payload)
+
+        // Clean the eventName up
+        var eventName = payload.event
+        if (EVENT_MAPPER.containsKey(eventName)) {
+            eventName = EVENT_MAPPER[eventName].toString()
+        } else {
+            eventName = makeKey(eventName)
+        }
+
+        payload.properties.let {
+            val formattedProperties = formatProperties(it)
+            firebaseAnalytics.logEvent(eventName, formattedProperties)
+            analytics.log("firebaseAnalytics.logEvent($eventName, $formattedProperties")
+        }
+
+        return returnPayload
+    }
+
+    override fun screen(payload: ScreenEvent): BaseEvent? {
+        var returnPayload = super.screen(payload)
+
+        val tempActivity = activity
+        if (tempActivity != null) {
+            firebaseAnalytics.setCurrentScreen(tempActivity, payload.name, null);
+        }
+
+        return returnPayload
+    }
+
+
+    // AndroidActivity Methods
+    override fun onActivityResumed(activity: Activity?) {
+        super.onActivityResumed(activity)
+
+        try {
+            val packageManager = activity?.packageManager ?: return
+
+            packageManager.getActivityInfo(activity.componentName, PackageManager.GET_META_DATA).let {
+                it.loadLabel(packageManager).toString().let {
+                    firebaseAnalytics.setCurrentScreen(activity, it, null)
+                    analytics.log("firebaseAnalytics.setCurrentScreen(activity, $it, null")
+                }
+            }
+        } catch (exception: PackageManager.NameNotFoundException) {
+            throw AssertionError("Activity Not Found: " + exception.toString())
+        }
+    }
+
+    override fun onActivityStarted(activity: Activity?) {
+        super.onActivityStarted(activity)
+        this.activity = activity
+    }
+
+    override fun onActivityStopped(activity: Activity?) {
+        super.onActivityStopped(activity)
+        this.activity = null
+    }
+
+    // Private Helper Methods
+
+    // Format properties into a format needed by firebase
+    private fun formatProperties(properties: Properties): Bundle? {
```

Comment:
Ref: https://developer.android.com/kotlin/ktx/extensions-list#androidxcoreos
`androidx.core:core-ktx` has a bunch of nicer APIs specifically for kotlin, to build Bundles. Could potentially look into using those to kotlin-ify the code a bit more 😄